### PR TITLE
Make value and comment as undefined when obs is hidden from control event

### PR DESCRIPTION
**Pull request by sowmika148**

### CODE REVIEW

File: src/helpers/ControlRecordWrapper.js [hunk 1]
```diff
@@ -66,11 +67,19 @@ export default class ControlRecordWrapper {
...
+
   setHidden(hidden) {
     const brotherTrees = ControlRecordTreeMgr.getBrothers(this.rootRecord, this.currentRecord);
 
     brotherTrees.forEach(r => {
-      const updatedRecord = r.set('hidden', hidden);
+      let updatedRecord = r.set('hidden', hidden);
+      if (hidden && this.isControlEvent()) {
+        const value = Object.assign(updatedRecord.value, { value: undefined, comment: undefined });
+        updatedRecord = updatedRecord.set('value', value);
```
Comment (shivarach) [inline src/helpers/ControlRecordWrapper.js]: Don't we need to do it for each brother tree recursively?

File: src/helpers/ControlRecordWrapper.js [hunk 2]
```diff
@@ -77,10 +77,27 @@ export default class ControlRecordWrapper {
...
+    return controlRecord.set('value', value);
+  }
+
+  clearObsValuesAndComment(controlRecord) {
+    let updatedRecord = controlRecord;
+    if (updatedRecord.control.type === 'obsControl') {
+      updatedRecord = this.setObsValueAndCommentToUndefined(updatedRecord);
+    } else if (updatedRecord.control.type === 'obsGroupControl'
+      || updatedRecord.control.type === 'table'
+      || updatedRecord.control.type === 'section') {
```
Comment (shivarach) [inline src/helpers/ControlRecordWrapper.js]: These explicit checks can be avoided by taking `updatedRecord.children` always and have a null check for map

**Review by shivarach — CHANGES_REQUESTED** [review]

File: src/helpers/ControlRecordWrapper.js (hunk 2 above)
Comment (sowmika148) [reply src/helpers/ControlRecordWrapper.js]: Done

File: src/helpers/ControlRecordWrapper.js (hunk 1 above)
Comment (sowmika148) [reply src/helpers/ControlRecordWrapper.js]: Done in next commit

**Review by shivarach — APPROVED** [review]

**Comment by shivarach** [comment]
LGTM

**Pull request merged**